x/concentrated-liquidity/lp.go

```diff
@@ -318,10 +318,10 @@ func (k Keeper) addToPosition(ctx sdk.Context, owner sdk.AccAddress, positionId
 	minimumAmount1 := amount1Withdrawn
 
 	if !amount0MinGiven.IsZero() {
-		minimumAmount0 = amount0MinGiven
+		minimumAmount0 = amount0Withdrawn.Add(amount0MinGiven)
```

Comment:
This should be the amount originally in the position prior to withdrawal, right? The actual withdrawn amount might vary due to rounding, so if e.g. `amount0MinGiven` is 1, then this would pass even if the final post-add position added 0

Reply:
Not sure if I follow here. Agreed that we could calculate based on the amount originally in the position, but what do you mean by "this would pass even if the final post-add position added 0"? @AlpinYukseloglu

Reply:
Exact case:
1. Starting position: 100 units of each asset, and we add 1 unit of each
2. `amount0MinGiven = 1`, meaning if we add less than 1 unit of asset0 we want to error. So any final result less than 101 units for asset0 should error.
3. Due to rounding/truncation, `amount0Withdrawn = 99`, meaning that `minimumAmount0 = 99 + 1 = 100`
4. This violates our expectation, since it sets the floor at `100` units, not `101`

Intuitively, if you have a 100 unit position and want to add at least 1 unit to it, your expected minimum outcome should be 101 units, not 100

Reply:
Not sure I follow here as well, I thought the consensus was to calculate based on the amount originally in the position?
If so, is it not opposed to statement number 2,

>  amount0MinGiven = 1, meaning if we add less than 1 unit of asset0 we want to error. So any final result less than 101 units for asset0 should error.

since we are expecting final result less than 1 unit of asset0 should error?

Maybe I did not understand the original issue correctly?

Reply:
Discussed offline & converged on replacing `amount0Withdrawn` here with the original amount in the position

Reply:
Re-discussed offline & converged to go as is for now